### PR TITLE
Add Makefile for development

### PR DESCRIPTION
**Pull request by f-moya**
Dante and I experienced some problems to get this working locally. Particularly while trying to install the [ffi](https://github.com/ffi/ffi) dependency. We experience an issue very simliar to [this one](https://github.com/ffi/ffi/issues/877#issue-783444672). It might be related to this repo using an old version of ruby `2.3`, but since updating the repo might gives some problems with Netlify, I found it easier to make it work using Docker.

### CODE REVIEW

**Comment by f-moya** [comment]
@filipesperandio I noticed the deploy for this failed with the following; which is something we have been warned about before [QUA-429](https://codeclimate.atlassian.net/browse/QUA-429).
```
10:03:13 PM: Build ready to start
10:03:13 PM: ---------------------------------------------------------------------
  UNSUPPORTED BUILD IMAGE

  The build image for this site uses Ubuntu 14.04 Trusty Tahr, which is no longer supported.

  To enable builds for this site, select a later build image at the following link:
  https://app.netlify.com/sites/cc-api-docs/settings/deploys#build-image-selection

  For more details, visit the build image migration guide:
  https://answers.netlify.com/t/end-of-support-for-trusty-build-image-everything-you-need-to-know/39004
```

You think updating that can be easy ? If you can help me getting access to Netlify I can take care of it. @dantevvp Will be opening a PR to add some docs soon.

**Comment by filipesperandio** [comment]
I pinged Jimmy on the ticket, I can't add you myself.

Thought, did a quick check and updating the image was just a click on a button.
The script is failing for other reasons now.

```
9:17:07 AM: Build ready to start
9:17:09 AM: build-image version: 195fbe127e5c374d9c4758652cb62e3b8936a395 (focal)
9:17:09 AM: build-image tag: v4.6.0
9:17:09 AM: buildbot version: 43c70c62bfed821cfaaabbb50f00238dd4bdd2d6
9:17:10 AM: Building without cache
9:17:10 AM: Starting to prepare the repo for build
9:17:10 AM: No cached dependencies found. Cloning fresh repo
9:17:10 AM: git clone https://github.com/codeclimate/slate
9:17:11 AM: Preparing Git Reference pull/18/head
9:17:11 AM: Parsing package.json dependencies
9:17:11 AM: Starting build script
9:17:11 AM: Installing dependencies
9:17:11 AM: Python version set to 2.7
9:17:12 AM: Downloading and installing node v10.24.1...
9:17:12 AM: Downloading https://nodejs.org/dist/v10.24.1/node-v10.24.1-linux-x64.tar.xz...
9:17:13 AM: Computing checksum with sha256sum
9:17:13 AM: Checksums matched!
9:17:14 AM: Now using node v10.24.1 (npm v6.14.12)
9:17:14 AM: Started restoring cached build plugins
9:17:14 AM: Finished restoring cached build plugins
9:17:15 AM: Required ruby-2.3.1 is not installed.
9:17:15 AM: To install do: 'rvm install "ruby-2.3.1"'
9:17:15 AM: Attempting ruby version 2.3.1, read from .ruby-version file
9:17:16 AM: Required ruby-2.3.1 is not installed - installing.
9:17:16 AM: Searching for binary rubies, this might take some time.
9:17:17 AM: Found remote file https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/20.04/x86_64/ruby-2.3.1.tar.bz2
9:17:17 AM: Checking requirements for ubuntu.
9:17:17 AM: Missing required packages: libssl1.0-dev
9:17:17 AM: RVM autolibs is now configured with mode '2' =>
9:17:17 AM:   'Allow RVM to use package manager if found, fail if dependencies are missing. This is default.',
9:17:17 AM: please run `rvm autolibs enable` to let RVM do its job or run and read `rvm autolibs [help]`
9:17:17 AM: or visit https://rvm.io/rvm/autolibs for more information.
9:17:17 AM: Found undesired packages: libssl-dev
9:17:17 AM: RVM autolibs is now configured with mode '2' =>
9:17:17 AM:   'Allow RVM to use package manager if found, fail if dependencies are missing. This is default.',
9:17:17 AM: please run `rvm autolibs enable` to let RVM do its job or run and read `rvm autolibs [help]`
9:17:17 AM: or visit https://rvm.io/rvm/autolibs for more information.
9:17:17 AM: Requirements installation failed with status: 1.
9:17:17 AM: ruby-2.3.1 - #gemset created /opt/buildhome/.rvm/gems/ruby-2.3.1
9:17:17 AM: Required ruby-2.3.1 is not installed - installing.
9:17:18 AM: Searching for binary rubies, this might take some time.
9:17:18 AM: Found remote file https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/20.04/x86_64/ruby-2.3.1.tar.bz2
9:17:18 AM: Checking requirements for ubuntu.
9:17:18 AM: Missing required packages: libssl1.0-dev
9:17:18 AM: RVM autolibs is now configured with mode '2' =>
9:17:18 AM:   'Allow RVM to use package manager if found, fail if dependencies are missing. This is default.',
9:17:18 AM: please run `rvm autolibs enable` to let RVM do its job or run and read `rvm autolibs [help]`
9:17:18 AM: or visit https://rvm.io/rvm/autolibs for more information.
9:17:18 AM: Found undesired packages: libssl-dev
9:17:18 AM: RVM autolibs is now configured with mode '2' =>
9:17:18 AM:   'Allow RVM to use package manager if found, fail if dependencies are missing. This is default.',
9:17:18 AM: please run `rvm autolibs enable` to let RVM do its job or run and read `rvm autolibs [help]`
9:17:18 AM: or visit https://rvm.io/rvm/autolibs for more information.
9:17:18 AM: Requirements installation failed with status: 1.
9:17:18 AM: ruby-2.3.1 - #importing gemsetfile /opt/buildhome/.rvm/gemsets/default.gems evaluated to empty gem list
9:17:19 AM: ruby-2.3.1 - #generating default wrappers.................
9:17:19 AM: Creating deploy upload records
9:17:19 AM: Error running 'run_gem_wrappers regenerate',
9:17:19 AM: please read /opt/buildhome/.rvm/log/1648124238_ruby-2.3.1/gemset.wrappers.default.log
9:17:19 AM: Using /opt/buildhome/.rvm/gems/ruby-2.3.1
9:17:19 AM: Warning! Executable 'ruby' missing, something went wrong with this ruby installation!
9:17:19 AM: Failed during stage 'building site': Build script returned non-zero exit code: 1 (https://ntl.fyi/exit-code-1)
9:17:19 AM: Warning! Executable 'gem' missing, something went wrong with this ruby installation!
9:17:19 AM: Warning! Executable 'irb' missing, something went wrong with this ruby installation!
9:17:19 AM: Using ruby version 2.3.1
9:17:19 AM: Using bundler version 1.14.5 from Gemfile.lock
9:17:19 AM: /opt/buildhome/.rvm/scripts/override_gem: line 19: gem: command not found
9:17:19 AM: Error installing bundler
9:17:19 AM: Build was terminated: Build script returned non-zero exit code: 1
9:17:19 AM: Failing build: Failed to build site
9:17:19 AM: Finished processing build request in 9.615454049s
```

The build command setup is there is `middleman build`.

File: README.md [hunk 1]
```diff
@@ -57,6 +57,12 @@ bundle exec middleman server
 
 # OR run this to run with vagrant
 vagrant up
```
Comment (filipesperandio) [inline README.md]: no body does vagrant these days, feel free to kill this.

**Comment by filipesperandio** [comment]
I think the failure is due to ruby version now. Seems like it uses the `.ruby-version` to determine, but the latest ubuntu (`focal`)  base image supports only ruby 2.6 and 2.7. Can we try using this version and updating `.ruby-version` in this branch?

https://docs.netlify.com/configure-builds/overview/#build-image-selection
https://github.com/netlify/build-image/blob/focal/included_software.md
https://github.com/netlify/build-image#available-images

**Comment by f-moya** [comment]
@filipesperandio Seems like it worked, I tried updating the dependencies as well but that gave me troubles so just updating the ruby version.

**Pull request merged**